src/v/cluster/partition_balancer_planner.cc

```diff
@@ -163,6 +163,61 @@ class partition_balancer_planner::request_context {
     ss::abort_source& _as;
 };
 
+/// Following two methods leak some of the space mgmt abstractions
+/// into the balancer code. That is something that needs fixing.
+/// The state is complicated because of upgrade considerations
+/// and whether the node space management has all the required
+/// information to send in a help report. These combinations
+/// are captured in a tristate + feature flag to account for
+/// all scenarios.
+
+uint64_t partition_balancer_planner::get_node_total_size_bytes(
+  const node::local_state& node_state) {
+    if (unlikely(!_state.is_trimming_space_management_active())) {
+        // in the middle of an upgrade, mixed mode
+        // fall back to disk based usage
+        return node_state.data_disk.total;
+    }
+    const auto& state = node_state.log_data_size;
+    if (state.is_disabled()) {
+        // feature is active but space management policies not in use
+        // fall back to disk based usage.
+        return node_state.data_disk.total;
+    } else if (state.has_optional_value()) {
+        return state.value().data_target_size;
+    }
+    // broker is still bootstrapping and information is
```

Comment:
Does this mean that the first few reports sent by a node are unusable? Why is this the case? We probably need to modify `partition_balancer_backend::on_health_monitor_update` as well then.

Reply:
> Does this mean that the first few reports sent by a node are unusable? Why is this the case?

This was the case before, because the fiber that computes the "free"/"usage" space runs async and may not be ready by the time first health report is dispatched. We made a small adjustment here by falling back to disk usage metrics in that window. That kind of simplified the code here.